### ISSUE TITLE
Define a JSON schema for the fluid files

### ISSUE THREAD

**Issue by ibell** [CoolProp/CoolProp#115]
Not for right now, but we can use JSON schema to fully describe the fluid files and standardize their form.  This makes the JSON files self-describing.

**Comment by thorade** [comment]
In my opinion, this is a great idea. Things would get even more interesting if various fluid properties libraries could agree upon a common schema.

**Comment by ibell** [comment]
Yes, that would be great, but I don't see that happening any time soon
unfortunately.  In any case, it would be great to define the fluid file
schema.  It's not a very trivial undertaking, but I did succeed for the
interaction parameters as an example.

**Comment by thorade** [comment]
A few ideas/comments:
- Most objects are on the second level, but some are on the top level (e.g. CAS and NAME). What do you think about introducing a new section INFO and moving all these there? Otherwise, it could be good (in terms or readability) to always have these top level objects at the top of the file.
- Might make sense to add a REVISIONS section.
- Maybe add a constant Dipole Moment.
- In addition to the FORMULA, one could add STRUCTUREFORMULA, e.g. `CH3-2(CH2)-CH3` for Butane.
- What is the convention for the NAME? Would it make sense to always use the IUPAC name (as in the info box on Wikipedia, eg.g https://en.wikipedia.org/wiki/Butane )?
- In addition to the CAS number one could also list the PubChem number or some other numbers. One could then link these numbers to some info pages, as it is done in the Wikipedia info box.
- For the references, one could list the DOI in addition to the Bibtex key. Or even use the DOI or shortDOI as the Bibtexkey? On the other hand keys like `Buecker-JPCRD-2006B` are easier to remember.
- In addition to the STATES given, one could give three more states: 
  - saturated liquid at T=273.15K (IIR)
  - saturated liquid at T=233.15K (ASHRAE)
  - saturated liquid at p=101325Pa (NBP)
- EOS now has an object a1,a2,reference,offset. Instead, one could just add a1 and a2 to the coefficients and move the reference state to the INFO section. Most ANCILLARIES and the STATES depend on the reference state anyway.
- If it should really happen that multiple libraries agree on a common schema, what about things that are not required for all libraries? E.g. the critical spline or hs_anchor are not used by other libaries. Are such things optional? How to agree on the handling of reference state?
- As I understand it, a schema would define e.g. what values are allowed for ANCILLARIES.rhoL.type. Documentation should then also explain what each value means.
- pure fluids pV and pL: see #711

**Comment by ibell** [comment]
I like a lot of these ideas.  Give me a chance to think a little about
them.  Are you interested in helping out with the schema?  That would be
enormously helpful to me.

**Comment by thorade** [comment]
I also see this as a long term goal, and I am interested in helping.
Maybe the work of defining a schema and the rigidity it introduces are only worth it if we find a third library to participate (besides CoolProp and HelmholtzMedia).

**Comment by ibell** [comment]
Do you want to try to open discussions with the ThermoC guys?  I'm swamped
at the moment, but I'd welcome your taking an active role in that.

**Comment by thorade** [comment]
OK, I'll contact the ThermoC guys.
